feat: EthL2s (#2092)
vegaprotocol/specs: protocol/0087-EL2D-ethereumL2-data-source.md

# Ethereum L2s data sources

## Summary 

This specification adds a new way of sourcing data from the Ethereum L2s to allow arbitrary data from the Ethereum L2s running EVM to be ingested as a [data source](./0045-DSRC-data_sourcing.md). 
This is in addition to and building upon [Ethereum data source](./0082-ETHD-ethereum-data-source.md).

## Decription 

In addition to listening to Ethereum events and reading from Ethereum contrants as described in [Ethereum data source](./0082-ETHD-ethereum-data-source.md) it will be possible for Vega nodes to listen to events from and read from other Ethereum L2s. 

The overarching principle is that the L2s incorporate the ethereum EVMs and thus contracts and ABIs are assumed to be functionally the same as on Ethereum itself. 

## Registration / removal

> A new network parameter, a JSDON list of network id / chain id / name one per L2 will be used for setting supported L2. This is set via [governance](./0028-GOVE-governance.md). 

Reviewer: ```suggestion
A new network parameter, a JSON list of network id / chain id / name one per L2 will be used for setting supported L2. This is set via [governance](./0028-GOVE-governance.md). 
```

Name `blockchains.ethereumL2Configs`.
Example value: 
```
{"configs": [{"network_id": 10, "chain_id": 10, "confirmations": 3, "name":"optimism"}]}
```

> Duplicate values of "network_id": 10, "chain_id" or  "name":"optimism" or not allowed (an update will be rejected at validation stage).

Reviewer: ```suggestion
Duplicate values of "network_id": 10, "chain_id" or  "name":"optimism" are not allowed (an update will be rejected at validation stage).
```

> Any updat emust always change the entire JSON (it's not possible to change individual entries).

Reviewer: ```suggestion
Any update must always change the entire JSON (it's not possible to change individual entries).
```
